patches/arcade/0004-Changing-GetLastStablePackage-logic-in-order-to-allo.patch

```diff
@@ -0,0 +1,233 @@
++                return null;
++
+             int build = version.Build, revision = version.Revision;
+ 
+             if (build == -1)
+diff --git a/src/Microsoft.DotNet.Build.Tasks.Packaging/src/build/Packaging.targets b/src/Microsoft.DotNet.Build.Tasks.Packaging/src/build/Packaging.targets
+index 06d45eff..156dbcd3 100644
+--- a/src/Microsoft.DotNet.Build.Tasks.Packaging/src/build/Packaging.targets
++++ b/src/Microsoft.DotNet.Build.Tasks.Packaging/src/build/Packaging.targets
+@@ -380,11 +380,16 @@
+       <_latestRuntimePackages Include="@(PkgProjDependency)" Condition="'%(PkgProjDependency.TargetRuntime)' != ''" KeepDuplicates="false" />
+     </ItemGroup>
+ 
++    <PropertyGroup>
++      <DoNotAllowVersionsFromSameRelease Condition="'$(DoNotAllowVersionsFromSameRelease)' == ''">true</DoNotAllowVersionsFromSameRelease>
++    </PropertyGroup>
++
+     <!-- Calculate the package version to harvest -->
+     <GetLastStablePackage Condition="'$(HarvestVersion)' == ''"
+                           LatestPackages="@(_latestPackage)"
+                           StablePackages="@(StablePackage)"
+-                          PackageIndexes="@(PackageIndex)">
++                          PackageIndexes="@(PackageIndex)"
++                          DoNotAllowVersionsFromSameRelease="$(DoNotAllowVersionsFromSameRelease)">
+       <Output  TaskParameter="LastStablePackages" ItemName="_lastStablePackage"/>
+     </GetLastStablePackage>
+ 
+@@ -396,14 +401,16 @@
+     <GetLastStablePackage Condition="'@(HarvestRuntimePackages)' == ''"
+                           LatestPackages="@(_latestRuntimePackages)"
+                           StablePackages="@(StablePackage)"
+-                          PackageIndexes="@(PackageIndex)">
++                          PackageIndexes="@(PackageIndex)"
++                          DoNotAllowVersionsFromSameRelease="$(DoNotAllowVersionsFromSameRelease)">
+       <Output  TaskParameter="LastStablePackages" ItemName="HarvestRuntimePackages"/>
+     </GetLastStablePackage>
+ 
+     <GetLastStablePackage Condition="'@(HarvestAdditionalPackageIds)' != ''"
+                           LatestPackages="@(HarvestAdditionalPackageIds)"
+                           StablePackages="@(StablePackage)"
+-                          PackageIndexes="@(PackageIndex)">
++                          PackageIndexes="@(PackageIndex)"
++                          DoNotAllowVersionsFromSameRelease="$(DoNotAllowVersionsFromSameRelease)">
+       <Output  TaskParameter="LastStablePackages" ItemName="HarvestAdditionalPackages"/>
+     </GetLastStablePackage>
+ 
+diff --git a/src/Microsoft.DotNet.Build.Tasks.Packaging/tests/GetLastStablePackageTests.cs b/src/Microsoft.DotNet.Build.Tasks.Packaging/tests/GetLastStablePackageTests.cs
+index aea43be5..3c818239 100644
+--- a/src/Microsoft.DotNet.Build.Tasks.Packaging/tests/GetLastStablePackageTests.cs
++++ b/src/Microsoft.DotNet.Build.Tasks.Packaging/tests/GetLastStablePackageTests.cs
```

Comment:
Do we need the tests included?

Reply:
We don't but this makes it easier in case other patches modify them later.  All of these new patches will go away when we uptake the new Arcade.